webpack/assets/javascripts/react_app/components/common/TypeAheadSelect/TypeAheadSelectActions.js

```diff
@@ -0,0 +1,31 @@
+import {
+  INIT,
+  UPDATE_OPTIONS,
+  UPDATE_SELECTED,
+} from './TypeAheadSelectConstants';
+import { mapSelected } from './TypeAheadSelectSelectors';
+
+export const initialUpdate = (options, selected, id) => ({
+  type: INIT,
+  payload: {
+    id,
+    options,
+    selected,
+  },
+});
+
+export const updateOptions = (options, id) => ({
+  type: UPDATE_OPTIONS,
+  payload: {
+    id,
+    options,
+  },
+});
+
+export const updateSelected = (selected, id) => ({
+  type: UPDATE_SELECTED,
+  payload: {
+    id,
+    selected: mapSelected(selected),
```

Comment:
I adding the mapping selector in the Action vs the Reducer because it actually only relates to updateSelected when called through the onChange handler